src/detection/battery/battery_apple.c

```diff
@@ -1,9 +1,34 @@
 #include "fastfetch.h"
 #include "battery.h"
 #include "util/apple/cf_helpers.h"
+#include "detection/temps/temps_apple.h"
 
 #include <IOKit/IOKitLib.h>
 
+static double detectBatteryTemp()
+{
+    FFlist temps;
+    ffListInit(&temps, sizeof(FFTempValue));
+
+    ffDetectCoreTemps(FF_TEMP_BATTERY, &temps);
+
+    if(temps.length == 0)
+        return FF_BATTERY_TEMP_UNSET;
+
+    double result = 0;
+    for(uint32_t i = 0; i < temps.length; ++i)
+    {
+        FFTempValue* tempValue = (FFTempValue*)ffListGet(&temps, i);
+        result += tempValue->value;
+        //TODO: do we really need this?
+        ffStrbufDestroy(&tempValue->name);
+        ffStrbufDestroy(&tempValue->deviceClass);
```

Comment:
I don't think this belongs here, as this is also the place where the list is being destroyed.

Reply:
I meant that although we got `name` and `deviceClass` but they were never displayed

Reply:
`name` and `deviceClass` are used on linux. 
`name` is used to identify which hwmon is the cpu, `deviceClass` is used by gpu, to map the device class found by libpci to the correct hwmon.

I don't think we need those on apple.